src/docbuilder/rustwide_builder.rs

```diff
@@ -69,6 +69,14 @@ const ESSENTIAL_FILES_UNVERSIONED: &[&str] = &[
 const DUMMY_CRATE_NAME: &str = "empty-library";
 const DUMMY_CRATE_VERSION: &str = "1.0.0";
 
+// This macro exists because of lifetimes issues surrounding the `Command::process_lines` method:
+// it expects a mutable reference to a closure, in which we capture a variable in order to store
+// the JSON output. Unfortunately, we *need* to create the command in the same scope level otherwise
+// rustc becomes very grumpy about the fact that the variable needs to be static because it is
+// captured in a mutably referenced closure.
```

Comment:
Would this be fixed by taking a closure by value instead? If so, let's open an issue on https://github.com/rust-lang/rustwide/ and put a TODO here to remove the macro once it's fixed.

Reply:
I honestly don't know but I would expect it to solve our issue.